packages/@react-stately/data/src/useListData.ts

```diff
@@ -211,9 +211,12 @@ export function createListActions<T>(opts: ListOptions<T>, dispatch: (updater: (
         let keySet = new Set(keys);
         let items = state.items.filter(item => !keySet.has(getKey(item)));
 
-        let selection = new Set(state.selectedKeys);
-        for (let key of keys) {
-          selection.delete(key);
+        let selection: Selection = 'all';
```

Comment:
If the collection becomes empty (i.e. removing all keys), should the selection change to an empty set? Looks like that's what happens in `removeSelectedItems` below.